src/App.jsx

```diff
@@ -4,8 +4,19 @@ import {BrowserRouter} from 'react-router-dom';
 import {Route, Switch, Redirect} from 'react-router-dom';
 import MainPage from './pages/MainPage';
 import MovieDetailsPage from './pages/MovieDetailsPage';
+import {useDispatch} from 'react-redux';
+import {getMoviesList} from './mocks/movies-list';
+import {useEffect} from 'react/cjs/react.development';
+import {Action} from './store/actions';
 
-function App() {
+const movies = getMoviesList();
+
+const App = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(Action.setMovies(movies));
+  }, movies);
```

Comment:
Зависимости в `useEffect` всегда массив. Но если твои данные - массив, то это не значит, что стоит его передавать в чистом виде. Нужно всегда передавать в зависимости массив с переменными. 

Это так же немного влияет и на производительность. В твоем массиве может быть тысяча элементов и `useEffect` будет бежать по всем и сверять с предыдущим состоянием. А так у тебя в зависимостях будет 1-3 переменные. По ним пробежаться быстро.

```suggestion
  }, [movies]);
```

Reply:
Исправлено.